src/graph/graphQL.ts

```diff
@@ -25,13 +29,185 @@ export function enumValue(value: string | string[]): ParameterEnum {
+ *  variables?: {[p: string]: string | boolean | number | ParameterEnum}}} options
+ * @returns {string}
+ */
+export function subscription(optionsOrName: SubscriptionOptions | string): string {
+    let options: SubscriptionOptions;
+
+    // Allow passing over a single string which would be the name of subscription file
+    if (typeof optionsOrName === "string") {
+        options = {
+            name: optionsOrName,
+        };
+    } else {
+        options = optionsOrName as SubscriptionOptions;
+    }
+
+    let s = options.subscription;
+    const fragmentDir = options.fragmentDir;
+    const path = options.path;
+    const name = options.name;
+    const t = trace.get();
+
+    // If subscription isn't defined attempt to load from file
+    if (!s) {
+        s = locateAndLoadGraphql({ path, name }, "subscription", t[1].getFileName());
+    }
+
+    // Replace variables
+    s = replaceParameters(s, options.variables);
+
+    // Inline fragments
+    s = inlineFragments(s, name, t[1].getFileName(), fragmentDir);
+
+    if (options.operationName) {
+        s = replaceOperationName(s, options.operationName);
+    }
+
+    // Inline entire subscription
+    if (options.inline !== false) {
+        s = inlineQuery(s);
+    }
+    return s;
+}
+
+export interface SubscriptionOptions {
+    subscription?: string;
+    path?: string;
+    name?: string;
+    fragmentDir?: string;
+    inline?: boolean;
+    variables?: {
+        [name: string]: string | boolean | number | ParameterEnum;
+    };
+    operationName?: string;
+}
+
+/**
+ * Prepare a GraphQL query string for the use with Apollo.
+ *
+ * Queries can be provided by the following options:
+ *
+ * * query: string containing the subscription GraphQL, or
+ * * path:  absolute or relative path to a .graphql file to load; if provided a relative
+ *          path this will resolve the relative path to an absolute given the location
+ *          of the calling script.
+ * * name:  name of the .graphql file to load; this will walk up the directory structure
+ *          starting at the location of the calling script and look for a folder called
+ *          'graphql'. Once that folder is found, by convention name is being looked for
+ *          in the 'query' sub directory.
+ * * fragmentsDir: location of fragment .graphql files
+ * * moduleDir: location of the calling script
+ * * inline: remove any unneeded whitespace and line breaks from returned GraphQL string
+ *
+ * @param {{query?: string; path?: string; name?: string; fragmentDir?: string; moduleDir: string; inline?: boolean}} options
+ * @returns {string}
+ */
+export function query<T, Q>(options: QueryOptions): string {
+    let q = options.query;
+    const fragmentDir = options.fragmentDir;
+    const path = options.path;
+    const name = options.name;
+
+    // If query isn't defined attempt to load from file
+    if (!q) {
+        q = locateAndLoadGraphql({ path, name }, "query", options.moduleDir);
```

Comment:
this one does not autopopulate the module dir from the stack trace?

Reply:
that's right, that method is really only used from within `ApolloGraphClient`. Ideally this would package protected. The moduleDir stuff is handled there.